### PR TITLE
Table row improvements

### PR DESCRIPTION
**Pull request by Ekrekr**
https://github.com/dataform-co/dataform-co/pull/6113

### CODE REVIEW

File: components/table.tsx [hunk 1]
```diff
@@ -1,41 +1,38 @@
 import { HTMLTable } from "@blueprintjs/core";
-import * as React from "react";
-
 import * as styles from "df/components/table.css";
+import * as React from "react";
 
-export type TRow = Array<{
-  cells: Array<React.ReactElement | string>;
+export interface IRow {
+  cells?: Array<React.ReactElement | string>;
```
Comment (GJMcGowan) [inline components/table.tsx:6]: Why can you have a row without cells?

File: components/table.tsx [hunk 2]
```diff
@@ -1,41 +1,38 @@
...
+export const Row = (
+  row: {
+    children: React.ReactElement;
+  } & IRow
+) => {
+  if (row.href || row.onClick) {
     return (
-      <a className={styles.tableRowLink} onClick={onClick} href={href}>
-        {children}
+      <a className={styles.tableRowLink} {...row}>
```
Comment (GJMcGowan) [inline components/table.tsx:29]: `...row` will add `colspans` to this tag, which is technically invalid. Easy way to remove is to extract colspans from row and then do `...rest`

File: components/table.tsx [hunk 3]
```diff
@@ -55,12 +52,12 @@ export const Table = ({ columnWidths, headers, rows, className, condensed }: ITa
       </tr>
     </thead>
     <tbody>
-      {rows.map(({ href, cells, colspans }, rowIndex) => (
-        <Row key={rowIndex} href={href}>
+      {rows.map((row, rowIndex) => (
+        <Row key={rowIndex} {...row}>
           <>
-            {cells.map((cell, cellIndex) => (
+            {row.cells.map((cell, cellIndex) => (
```
Comment (GJMcGowan) [inline components/table.tsx:58]: if cells is optional now this should be `row?.cells`

File: components/table.tsx (hunk 1 above)
Comment (Ekrekr) [reply components/table.tsx]: Ah you shouldn't be able to, nice spot

File: components/table.tsx (hunk 2 above)
Comment (Ekrekr) [reply components/table.tsx]: Nice

**Pull request merged**